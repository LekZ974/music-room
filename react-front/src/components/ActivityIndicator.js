import React from 'react';
import { ActivityIndicator as ActivityIndicatorComponent, StyleSheet } from 'react-native';
import PropTypes from 'prop-types';

const styles = StyleSheet.create({
  loading: {
    position: 'absolute',
    left: 0,
    right: 0,
    top: 0,
    bottom: 0,
    flex: 1,
    alignItems: 'center',
    justifyContent: 'center',
  },
  bgColor: {
    backgroundColor: '#ffffff',
  },
});

const ActivityIndicator = ({ animating, transparent }) => {
  return (
    <ActivityIndicatorComponent
      style={[styles.loading, !transparent && styles.bgColor]}
      animating={animating}
      size="large"
    />
  );
};

ActivityIndicator.defaultProps = {
  transparent: false,
};

ActivityIndicator.propTypes = {
  animating: PropTypes.bool.isRequired,
  transparent: PropTypes.bool,
};

export default ActivityIndicator;
