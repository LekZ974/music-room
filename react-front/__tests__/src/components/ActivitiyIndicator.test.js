import React from 'react';
import renderer from 'react-test-renderer';
import ActivityIndicator from '../../../src/components/ActivityIndicator';

it('renders correctly', () => {
  const tree = renderer.create(<ActivityIndicator animating />).toJSON();

  expect(tree).toMatchSnapshot();
});
