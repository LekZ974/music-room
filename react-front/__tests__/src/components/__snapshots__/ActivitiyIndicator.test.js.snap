// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<ActivityIndicator
  animating={true}
  color="#999999"
  hidesWhenStopped={true}
  size="large"
  style={
    Array [
      Object {
        "alignItems": "center",
        "bottom": 0,
        "flex": 1,
        "justifyContent": "center",
        "left": 0,
        "position": "absolute",
        "right": 0,
        "top": 0,
      },
      Object {
        "backgroundColor": "#ffffff",
      },
    ]
  }
/>
`;
